### PR TITLE
Fix DDIM for eta>0 and a cpu generator

### PR DESCRIPTION
**Pull request by anton-l**
Follow-up to https://github.com/huggingface/diffusers/pull/1189
Allows DDIM to still be used with a cpu generator and eta>0 in `DDIMPipeline`

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_ddim.py [hunk 1]
```diff
@@ -300,10 +300,14 @@ def step(
...
                     variance_noise = variance_noise.to(device)
-                else:
+                elif generator is not None and generator.device.type == device.type:
                     variance_noise = torch.randn(
                         model_output.shape, generator=generator, device=device, dtype=model_output.dtype
                     )
+                else:
+                    variance_noise = torch.randn(model_output.shape, generator=generator, dtype=model_output.dtype).to(
+                        device
+                    )
```
Comment (patil-suraj) [inline src/diffusers/schedulers/scheduling_ddim.py:310]:
```suggestion
                    variance_noise = torch.randn(model_output.shape, generator=generator, dtype=model_output.dtype)
                        variance_noise = variance_noise.to(device)

```

**Review by patil-suraj — APPROVED** [review]
LGTM, thanks!

**Comment by pcuenca** [comment]
MPS tests fail, do you want me to take a look?

**Comment by anton-l** [comment]
@pcuenca yeah, I think those accumulated over the past two days :sweat_smile:

**Comment by anton-l** [comment]
Superseded by https://github.com/huggingface/diffusers/pull/1222

**Pull request closed without merging**